framework/include/materials/MaterialPropertyInterface.h

```diff
@@ -460,6 +499,14 @@ template <>
 const MaterialProperty<Real> *
 MaterialPropertyInterface::defaultMaterialProperty<Real>(const std::string & name);
 
+template <>
+const FunctorMaterialProperty<Real> *
+MaterialPropertyInterface::defaultFunctorMaterialProperty<Real>(const std::string & name);
+
+template <>
+const FunctorMaterialProperty<ADReal> *
+MaterialPropertyInterface::defaultFunctorMaterialProperty<ADReal>(const std::string & name);
+
 template <>
 const ADMaterialProperty<Real> *
 MaterialPropertyInterface::defaultADMaterialProperty<Real>(const std::string & name);
```

Comment:
can you re-order this?
non AD
AD
non AD
AD
like the others

Reply:
Isn't this

non-AD
AD

right now?

Reply:
right now it goes 
NonAD
NonAD
AD
AD
..

Reply:
```c++
template <>
const FunctorMaterialProperty<Real> *
MaterialPropertyInterface::defaultFunctorMaterialProperty<Real>(const std::string & name);

template <>
const FunctorMaterialProperty<ADReal> *
MaterialPropertyInterface::defaultFunctorMaterialProperty<ADReal>(const std::string & name);

template <>
const ADMaterialProperty<Real> *
MaterialPropertyInterface::defaultADMaterialProperty<Real>(const std::string & name);
```
It goes non-AD and then AD for the `defaultFunctorMaterialProperty` method instantiations and then the next method is `defaultADMaterialProperty` (so different method), so I think this is fine?

Reply:
if you feel like catering to the OCD there, it currently goes:
routine 1 non AD
routine 2 non AD
routine 2 AD
routine 1 AD
routine 3 non AD
routine 3 AD

Obviously this is not stopping this PR (I think the caching is rn?)
```
// Forward declare explicit specializations
template <>
const MaterialProperty<Real> *
MaterialPropertyInterface::defaultMaterialProperty<Real>(const std::string & name);

template <>
const FunctorMaterialProperty<Real> *
MaterialPropertyInterface::defaultFunctorMaterialProperty<Real>(const std::string & name);

template <>
const FunctorMaterialProperty<ADReal> *
MaterialPropertyInterface::defaultFunctorMaterialProperty<ADReal>(const std::string & name);

template <>
const ADMaterialProperty<Real> *
MaterialPropertyInterface::defaultADMaterialProperty<Real>(const std::string & name);

template <>
const MaterialProperty<RealVectorValue> *
MaterialPropertyInterface::defaultMaterialProperty<RealVectorValue>(const std::string & name);
template <>
const ADMaterialProperty<RealVectorValue> *
MaterialPropertyInterface::defaultADMaterialProperty<RealVectorValue>(const std::string & name);

```

Reply:
K I think this should appeal to the OCD now...

Reply:
Please mark as resolved @GiudGiud if you're satisfied